app/src/main/java/com/github/geohunt/app/ui/components/activehunts/ChallengePreview.kt

```diff
@@ -0,0 +1,146 @@
+ */
+@Composable
+fun ChallengeImage(challenge: Challenge, modifier: Modifier) {
+    val thumbnail = challenge.thumbnail
+
+    Box(modifier = modifier.fillMaxWidth()) { //image "frame"
+        FetchComponent(lazyRef = { thumbnail }) {resolvedThumbnail ->
+            Image(painter = BitmapPainter(resolvedThumbnail.asImageBitmap()),
+                    contentDescription = "Challenge ${challenge.cid}",
+                    modifier = Modifier.clip(RoundedCornerShape(20.dp)).fillMaxSize(),
+                    contentScale = ContentScale.FillBounds)
+        }
+    }
+}
+
+/**
+ * A basic description of the challenge,
+ * displays the author, the location and the
+ * expiration date
+ */
+@Composable
+fun ChallengeDescription(challenge: Challenge, modifier: Modifier) {
+    Column(modifier = modifier, verticalArrangement = Arrangement.SpaceBetween) {
+        AuthorName(challenge.author)
+
+        Location()
+
+        ExpirationTime(expirationDate = challenge.expirationDate)
+    }
+}
+
+@Composable
+fun AuthorName(author: LazyRef<User>) {
+    val waitingAuthor = rememberLazyRef { author }
+    Row(modifier = Modifier.fillMaxWidth()) {
+        Icon(Icons.Rounded.Person, contentDescription = "person_icon")
+
+        Spacer(modifier = Modifier.size(10.dp))
+
+        if(waitingAuthor.value != null) Text(text = author.value!!.name)
+        else Text(text = "...")
+    }
+}
+
+@Composable
+fun Location(){
+    //TODO : Hardcoded values, should be replaced with a way to find closest city
+    //or removed entirely
+    val city = "Rome"
+    val country = "Italy"
+
+    Row(modifier = Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
+        Icon(Icons.Rounded.LocationOn,
+        contentDescription = "location_icon")
+
+        Spacer(modifier = Modifier.size(10.dp))
+
+        Text(
+                buildAnnotatedString {
+                    withStyle(style = SpanStyle(fontWeight = FontWeight.SemiBold)) {
+                        append(city)
+                    }
+                    append(", $country")
+                })
+
+    }
+}
+
+@Composable
+fun ExpirationTime(expirationDate: LocalDateTime?) {
+    val expires = if (expirationDate != null) "Expires in " else "Expires "
+    val expirationDateFmt = DateFormatUtils.formatRemainingTime(expirationDate).lowercase()
+
+    Row(modifier = Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
+        Icon(Icons.Rounded.CalendarMonth,
+                contentDescription = "calendar_icon")
+
+        Spacer(modifier = Modifier.size(10.dp))
+
+        Text(buildAnnotatedString {
+            withStyle(style = SpanStyle(fontWeight = FontWeight.SemiBold)) {
+                append(expires)
+
+                withStyle(style = SpanStyle(color = geoHuntRed)) {
+                    append(expirationDateFmt)
+                }
+
+                append(" !")
+            }
+
+        })
+    }
+}
```

Comment:
If you take the last commit on my branch we could merge your expiration time format with mine (as such resolving the comment of @Maeeen)